Fix decode error from possible image mode mismatch

SiriusStarr opened the pull request:
Changed call to PIL.Image.frombytes to always use mode='RGBA' rather
than determining mode from texture.colorfmt.  texture.pixels always
returns data as RGBA, so no other mode is ever appropriate to use.

Associated issue #41

SiriusStarr committed:
Fix decode error from possible image mode mismatch

Changed call to PIL.Image.frombytes to always use mode='RGBA' rather
than determining mode from texture.colorfmt.  texture.pixels always
returns data as RGBA, so no other mode is ever appropriate to use.

SiriusStarr committed:
Fix erroneous failure in test

Test suite expected no codes found when in fact two should have been.
Updated test to expect the correct two codes to be returned.

SiriusStarr commented:
The failing test is due to a bug in the test suite.  Per the comment:
```python
# currently detects no codes, but that's a bug
self.assertEqual(symbols, [])
```
should be
```python
self.assertEqual(
    symbols, [
        ZBarCam.Symbol(type='QRCODE', data=b'zbarlight test qr code'),
        ZBarCam.Symbol(type='UPCA', data=b'012345678905')
    ]
)
```

I've updated the test suite accordingly in the pull request, since it now works (was presumably failing due to the RGBA mode bug in question?).

AndreMiras commented on zbarcam/zbarcam.py:
```diff
@@ -91,13 +91,7 @@ def _on_texture(self, instance):
-        if cls.is_ios() and fmt == 'BGRA':
-            fmt = 'RGBA'
-        pil_image = PIL.Image.frombytes(mode=fmt, size=size, data=image_data)
+        pil_image = PIL.Image.frombytes(mode='RGBA', size=size, data=image_data)
```
Minor: Flake8 is complaining, this line is one character too long :grimacing: 
```
zbarcam/zbarcam.py:94:80: E501 line too long (80 > 79 characters)
```

AndreMiras replied:
Also would you mind putting a code comment and linking to #41 explaining why we're doing forcing the mode to RGBA.
That would save people time when going through the code later on

AndreMiras commented:
Great research and work, thank you!. This actually also is related to @GoBig87 initial work and findings https://github.com/kivy-garden/garden.zbarcam/pull/25 
I'll do some tests later this week.
In the mean time would you mind addressing the couple of minor comments I left on the PR?

SiriusStarr committed:
Fix linter warning and explain changes

Shortened line that was too long per PEP 8.

Added explanatory comment for change to forcing 'RGBA' mode with link
to issue #41.

SiriusStarr commented:
Updated for line length (I'm bad and use 100, so I didn't notice) and added a comment referencing #41 .

AndreMiras commented:
Thanks for the update :pray: 
Merging :+1:
